scanpipe/pipes/scancode.py

```diff
@@ -436,6 +442,25 @@ def assemble_packages(project):
                     logger.info(f"Unknown Package assembly item type: {item!r}")
 
 
+def get_packages_with_purl_from_resources(project):
+    """
+    Yield Dependency or PackageData objects created from detected package_data
+    in all the project resources. Both Dependency and PackageData objects have
+    the `purl` attribute with a valid purl.
+    """
+    for resource in project.codebaseresources.has_package_data():
+        for package_mapping in resource.package_data:
+            for dep in package_mapping.get("dependencies"):
+                yield packagedcode_models.Dependency.from_dependent_package(
+                    dependent_package=dep,
+                    datafile_path=resource.path,
+                    datasource_id=package_mapping.get("datasource_id"),
+                    package_uid=None,
+                )
+            pd = packagedcode_models.PackageData.from_dict(mapping=package_mapping)
+            yield pd
```

Comment:
Assigning to a var is not useful here unless the var name helps with the readability, which is not the case.
```suggestion
            yield packagedcode_models.PackageData.from_dict(mapping=package_mapping)
```

Reply:
Done, thanks!